Libraries/config/Headers/config/Config.h

```diff
@@ -0,0 +1,39 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __config_config_h
+#define __config_config_h
+
+#include <libutil/Filesystem.h>
+#include <plist/Object.h>
+
+namespace config {
+
+class Config {
+public:
+    Config(libutil::Filesystem const *filesystem, std::string const &defaultConfigPath = "/var/db/xcbuild_defaults.plist");
```

Comment:
One thing to consider would be making this object just hold the *data* from the config, and making a static method like `Config::Load()` to actually load the configuration. So the constructor here would take `Config(std::vector<std::string> const &extraPlatformPaths, std::vector<std::string> const &extraToolchainPaths)` and just hold the data.

That way, you could easily create a config with the data you want, but it's still just one call to load the data. It also makes it easy for loading to fail — unlike a constructor, the `Load()` method could return `ext::optional<Config>` and `ext::nullopt` on failure.